Source/TBS_Game/Private/Field/Building/Building.cpp

```diff
@@ -1,22 +1,79 @@
 #include "Field/Building/Building.h"
-#include "Field/Anchor/CellParamsMap.h"
+
+#include <Engine/ActorChannel.h>
+#include <Net/UnrealNetwork.h>
+#include "Field/Anchor/CellParamsMapGenerator.h"
 #include "Field/Building/BuildingView.h"
 #include "Field/Building/UpgradeBuildingComponent.h"
 #include "Field/Controller/FieldController.h"
-#include "Field/Event/TurnsOrderEventSystem.h"
 #include "Field/ReturnState/BuildingPlacementReturnState.h"
 #include "Field/ReturnState/BuildUpgradeReturnState.h"
 #include "Player/GamePlayerController.h"
 #include "Field/FieldActor.h"
+#include "Field/Anchor/CellParametersType.h"
+#include "Utils/TwoDimArray/CellParamsTwoDimArray.h"
 
 
 void ABuilding::Init(AFieldController* Field, AGamePlayerController* PlayerControllerOwner)
 {
-	SetOwner(PlayerControllerOwner);
 	PlayerControllerRef = PlayerControllerOwner;
 	AGameActor::Init(Field);
 }
 
+void ABuilding::GetLifetimeReplicatedProps(TArray<FLifetimeProperty>& OutLifetimeProps) const
+{
+	Super::GetLifetimeReplicatedProps(OutLifetimeProps);
+	// General properties
+	DOREPLIFETIME( ABuilding, UpgradeBuildingComponents );
```

Comment:
Can we have an array of properties and call `DOREPLIFETIME` with this properties?